nsqd/protocol_v2.go

```diff
@@ -62,7 +62,11 @@ func (p *protocolV2) IOLoop(conn net.Conn) error {
 		// ie. the returned slice is only valid until the next call to it
 		line, err = client.Reader.ReadSlice('\n')
 		if err != nil {
-			err = fmt.Errorf("failed to read command - %s", err)
+			if err == io.EOF && atomic.LoadInt32(&client.State) == stateClosing {
```

Comment:
I'm not so sure that ErrBufFull should be ignored, but since we're closing I guess it really *doesn't* matter.